File: _java/symulacja_organizmow/src/Symulacja/Zwierzeta/CyberOwca.java
package Symulacja.Zwierzeta;
import Symulacja.*;

import java.awt.*;

public class CyberOwca extends Zwierze {
    protected void SetInicjatywa(){
        this.inicjatywa=4;
    };
    protected void SetSila(){
        this.sila=11;
    };
    public CyberOwca(Swiat swiat){
        super(swiat);
        SetInicjatywa();
        SetSila();
    };
    public char Rysowanie(){
        return 'C';
    };
    public Organizm Urodz() {
        return new CyberOwca(this.swiat);
    };
    public boolean MozeBycPartnerem(Organizm organizm){
        if(organizm instanceof CyberOwca)
        return true;
    else
        return false;
    };
    @Override
    public Color GetColor(){
        return new Color(100,0,0);
    };
}

File: _c++/symulacja_organizmow/Organizmy/Zwierzeta/CyberOwca.cpp
//
//  CyberOwca.cpp
//  WirtualnySwiat
//
//  Created by Michal Tarnacki on 10/04/2022.
//

#include "CyberOwca.hpp"

CyberOwca::CyberOwca(Swiat &swiat):Zwierze(swiat){
    SetInicjatywa();
    SetSila();
};
void CyberOwca::Akcja(){
    Zwierze::Akcja();
}
void CyberOwca::Kolizja(Organizm& organizm){
    Zwierze::Kolizja(organizm);
}
char CyberOwca::Rysowanie() const{
    return 'C';
}
void CyberOwca::SetInicjatywa(){
    this->inicjatywa=4;
}
void CyberOwca::SetSila(){
    this->sila=11;
}
Organizm* CyberOwca::Urodz(){
    return new CyberOwca(this->swiat);
}
bool CyberOwca::MozeBycPartnerem(Organizm& organizm){
    if(dynamic_cast<CyberOwca*>(&organizm))
       return true;
    else
       return false;
}
